mongo/datadog_checks/mongo/mongo.py

```diff
@@ -83,18 +81,73 @@ class MongoDb(AgentCheck):
         10: ('REMOVED', 'Removed'),
     }
 
-    def __init__(self, name, init_config, agentConfig, instances=None):
-        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+    def __init__(self, name, init_config, instances=None):
+        super(MongoDb, self).__init__(name, init_config, instances)
 
         # Members' last replica set states
         self._last_state_by_server = {}
 
-        # List of metrics to collect per instance
-        self.metrics_to_collect_by_instance = {}
+        self.collection_metrics_names = (key.split('.')[1] for key in metrics.COLLECTION_METRICS)
```

Comment:
This change introduced a bug as it converted the list into a generator that can only be iterated once. I thought this was creating a tuple